### BWA 
For official website, read [Burrows-Wheeler Aligner](https://bio-bwa.sourceforge.net/). <br>
For official article, read [Heng Li , Richard Durbin, Fast and accurate short read alignment with Burrows–Wheeler transform, Bioinformatics, Volume 25, Issue 14, July 2009, Pages 1754–1760](https://academic.oup.com/bioinformatics/article/25/14/1754/225615?login=false).

### CD-HIT
For official website, read [CD-HIT](https://sites.google.com/view/cd-hit). <br>
For running the program, read [User's Guide](https://github.com/weizhongli/cdhit/wiki/3.-User's-Guide). <br>
For official article, read ["Clustering of highly homologous sequences to reduce the size of large protein database", Weizhong Li, Lukasz Jaroszewski & Adam Godzik Bioinformatics, (2001) 17:282-283](https://academic.oup.com/bioinformatics/article/17/3/282/189639?login=true).

### Clustal omega
For official website, read [Clustal Omega](http://www.clustal.org/omega/). <br>
For official article, read [Sievers, Fabian, Wilm, Andreas, Dineen, David, Gibson, Toby J, Karplus, Kevin, Li, Weizhong, Lopez, Rodrigo, McWilliam, Hamish, Remmert, Michael, Söding, Johannes, Thompson, Julie D, Higgins, Desmond G, (2011) Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega. Molecular Systems Biology, 7. 539](https://www.embopress.org/doi/full/10.1038/msb.2011.75). 

### FastANI
For official website, read [FastANI](https://github.com/ParBLiSS/FastANI). <br>
For official article, read [Jain, C., Rodriguez-R, L.M., Phillippy, A.M. et al. High throughput ANI analysis of 90K prokaryotic genomes reveals clear species boundaries. Nat Commun 9, 5114 (2018)](https://www.nature.com/articles/s41467-018-07641-9).

### FastQC 
For official website, read [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/). <br>
For manual, type ``fastqc -h``. 

### Glimmer
For official website, read [Glimmer](http://ccb.jhu.edu/software/glimmer/index.shtml). <br>
For official article, read [S. Salzberg, A. Delcher, S. Kasif, and O. White. Microbial gene identification using interpolated Markov models, Nucleic Acids Research 26:2 (1998), 544-548](https://academic.oup.com/nar/article/26/2/544/1174019). 

### IQ-TREE
For official website, read [IQ-TREE](http://www.iqtree.org/). <br>
For official article, read [Lam-Tung Nguyen and others, IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies, Molecular Biology and Evolution, Volume 32, Issue 1, January 2015, Pages 268–274](https://academic.oup.com/mbe/article/32/1/268/2925592?login=true). 

### MultiQC
For official website, read [MultiQC](https://multiqc.info/). <br>
For official article, read [Philip Ewels and others, MultiQC: summarize analysis results for multiple tools and samples in a single report, Bioinformatics, Volume 32, Issue 19, October 2016, Pages 3047–3048](https://academic.oup.com/bioinformatics/article/32/19/3047/2196507).

### MUMmer 
For official website, read [Ultra-fast alignment of large-scale DNA and protein sequences](https://mummer.sourceforge.net/). <br>
For official article, read [Arthur L. Delcher and others, Alignment of whole genomes, Nucleic Acids Research, Volume 27, Issue 11, 1 June 1999, Pages 2369–2376](https://academic.oup.com/nar/article/27/11/2369/1258501?login=false). 

### Qualimap
For official website, read [Qualimap 2.3 documentation](http://qualimap.conesalab.org/doc_html/index.html). <br>
For official article, read [Fernando García-Alcalde and others, Qualimap: evaluating next-generation sequencing alignment data, Bioinformatics, Volume 28, Issue 20, October 2012, Pages 2678–2679](https://academic.oup.com/bioinformatics/article/28/20/2678/206551).

### Quast
For official website, read [Quast](https://quast.sourceforge.net/). <br>
For official article, read [Alexey Gurevich and others, QUAST: quality assessment tool for genome assemblies, Bioinformatics, Volume 29, Issue 8, April 2013, Pages 1072–1075](https://academic.oup.com/bioinformatics/article/29/8/1072/228832).

### Roary
For official website, read [Roary: the Pan Genome Pipeline](http://sanger-pathogens.github.io/Roary/). <br>
For official article, read [Andrew J. Page and others, Roary: rapid large-scale prokaryote pan genome analysis, Bioinformatics, Volume 31, Issue 22, November 2015, Pages 3691–3693](https://academic.oup.com/bioinformatics/article/31/22/3691/240757).

### Sealer
For official article, read [Paulino, D., Warren, R.L., Vandervalk, B.P. et al. Sealer: a scalable gap-closing application for finishing draft genomes. BMC Bioinformatics 16, 230 (2015)](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0663-4).

### SPAdes
For official website, read [SPAdes](https://cab.spbu.ru/software/spades/). <br>
Alekseyev M. A. SPAdes: новый ассемблер геномов с поддержкой одноклеточного секвенирования. (talk), 2011. <br>
For official artical, read [Bankevich A, Nurk S, Antipov D, Gurevich AA, Dvorkin M, Kulikov AS, Lesin VM, Nikolenko SI, Pham S, Prjibelski AD, Pyshkin AV, Sirotkin AV, Vyahhi N, Tesler G, Alekseyev MA, Pevzner PA. SPAdes: a new genome assembly algorithm and its applications to single-cell sequencing. J Comput Biol. 2012 May;19(5):455-77](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3342519/). 

### Spine 
For offical website, read [Spine](https://github.com/egonozer/Spine). <br>
For official article, read [Ozer, E.A., Allen, J.P. & Hauser, A.R. Characterization of the core and accessory genomes of Pseudomonas aeruginosa using bioinformatic tools Spine and AGEnt. BMC Genomics 15, 737 (2014)](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-15-737). 

### Trimmomatic
For official website, read [Trimmomatic: A flexible read trimming tool for Illumina NGS data](http://www.usadellab.org/cms/?page=trimmomatic). <br> 
For official article, read [Anthony M. Bolger and others, Trimmomatic: a flexible trimmer for Illumina sequence data, Bioinformatics, Volume 30, Issue 15, August 2014, Pages 2114–2120](https://academic.oup.com/bioinformatics/article/30/15/2114/2390096). 

### Prokka
For official website, read [Prokka: rapid prokaryotic genome annotation](https://github.com/tseemann/prokka). <br>
For official artical, read [Torsten Seemann, Prokka: rapid prokaryotic genome annotation, Bioinformatics, Volume 30, Issue 14, July 2014, Pages 2068–2069](https://academic.oup.com/bioinformatics/article/30/14/2068/2390517).












